schematic/manifest/generator.py

```diff
@@ -1605,22 +1609,20 @@ def populate_existing_excel_spreadsheet(self, existing_excel_path: str = None, a
         out_of_schema_columns_lst = list(out_of_schema_columns)
         
         # initalize excel writer
-        writer = pd.ExcelWriter(existing_excel_path, engine='openpyxl')
-        writer.worksheets = {ws.title: ws for ws in workbook.worksheets}
-        worksheet = writer.worksheets["Sheet1"]
+        with pd.ExcelWriter(existing_excel_path, mode='a', engine='openpyxl', if_sheet_exists='overlay') as writer: 
 
-        # add additional content to the existing spreadsheet
-        additional_df.to_excel(writer, sheet_name = "Sheet1", startrow=1, index = False, header=False)
+            writer.worksheets = {ws.title: ws for ws in workbook.worksheets}
+            worksheet = writer.worksheets["Sheet1"]
 
-        # if there are new columns, add them to the end of spreadsheet
-        if len(out_of_schema_columns_lst) > 0:
-            for i, col_name in enumerate(out_of_schema_columns_lst):
-                col_index = len(workbook_headers) + 1 + i
-                worksheet.cell(row=1, column=col_index).value = col_name
-        
-        # save and close
-        writer.save()
-        writer.close()
+            # if there are new columns, add them to the end of spreadsheet
+            # Note: previously, we tried looping through the out of schema columns and use worksheet.cell to modify a particular cell. But that functionality is no longer working. 
+            if len(out_of_schema_columns_lst) > 0:
```

Comment:
Nit: Python has implicit booleanness.  [stackoverflow](https://stackoverflow.com/questions/53513/how-do-i-check-if-a-list-is-empty)

That said, I think it is fine to explicitly call this the `len() > 0`.  I just personally like to leverage the implicit booleanness which is "more Pythonic" `if out_of_schema_columns_lst`

Reply:
Thanks Tom. I like your approach. I will update the code to reflect it.